### ISSUE TITLE
Clear icon doesn't show when readOnly is true

### ISSUE THREAD

**Issue by adamgruber** [Hacker0x01/react-datepicker#422]
So I've set up a datepicker inside a form. I want my users to be able to clear the picker so I set `isClearable: true` and I get the close icon. Yay! :smile: I would also like to disable typing in the input box so I set `readOnly: true`. Now the close icon no longer appears. Boo! :cry: 

Any way to enable clearing of the input box while also preventing users from typing into it?

Honestly I wouldn't mind if users could type into the box and have it recognized but the fact that you type in the box and then it just disappears is not the best experience which is why I'd like to disable typing.

**Comment by rafeememon** [comment]
What version are you running?  Setting `isClearable` and `readOnly` seems to work as expected for me.

**Comment by adamgruber** [comment]
I'm using version `0.24.0`. I looked a little deeper and I do see that the close icon is indeed getting rendered but it is being hidden by this line of CSS:
```css
.datepicker__input:not(:valid)~.close-icon {
  display: none;
}
```

**Comment by rafeememon** [comment]
Ah, turns out this was unwittingly fixed in #412.  @martijnrusschen, we should probably cut a release to include this and other recent fixes

**Comment by martijnrusschen** [comment]
Released 0.25, this should fix the issue I think. https://github.com/Hacker0x01/react-datepicker/releases/tag/v0.25.0
